src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -91,7 +95,58 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
 
     componentDidMount(){
 
+    }
+
+    componentDidUpdate() {
+        this.addBookmarkButtons();
+    }
+
+    public addBookmarkButtons() {
+        let shareURL = window.location.href;
+
+        if (!shareURL.includes("session_id")) return;
+
+        const showSamples = shareURL.indexOf("&show");
+        if (showSamples > -1) {
+            shareURL = shareURL.slice(0, showSamples);
+        }
+
+        if (shareURL.includes('localhost')) {
+            this.addAddThis(shareURL);
+        } else {
+            request
+                .get('http://' + location.host + "/api/url-shortener?url=" + shareURL)
+                .end((err, res) => {
+                    if (err || !res.ok) {
+                        this.addAddThis(shareURL);
+                    } else {
+                        this.addAddThis(res.shortURL);
+                    }
+                });
+        }
+    }
 
+    public addAddThis(shareURL:string) {
+        const showTwitter = Boolean(AppConfig.showTwitter) === true;
+        const twitterParam = showTwitter ? {
+            passthrough: {
+                twitter: {
+                    text: "#cbioportal",
+                    url: shareURL
+                }
+            }
+        } : {};
+        addthis(
+            {pubid: 'ra-59959e35901cdde7'},
+            {url: shareURL, ...twitterParam},
+            () => {
+                const addthisDiv = document.createElement("div");
+                Object.assign(addthisDiv.style, {display: 'flex', justifyContent: 'flex-end'});
+                if (showTwitter) addthisDiv.classList.add("addthis_inline_share_toolbox");
+                else addthisDiv.classList.add("addthis_inline_share_toolbox_ubww");
+                document.getElementsByClassName("main_smry")[0].appendChild(addthisDiv);
```

Comment:
We want to avoid this kind of DOM reference.  Instead, we should use a component which can manage its own dom and call addthis against it.